src/v/cluster/health_monitor_types.cc

```diff
@@ -101,6 +106,51 @@ std::ostream& operator<<(std::ostream& o, const partition_status& ps) {
     return o;
 }
 
+topic_status& topic_status::operator=(const topic_status& rhs) {
+    if (this == &rhs) {
```

Comment:
Clang tidy was giving my warnings about self assignment, which I don't fully understand because I believe I'm handling it correctly below, but that's where this is coming from.